grype/pkg/syft_sbom_provider.go

```diff
@@ -23,67 +34,120 @@ func (e errEmptySBOM) Error() string {
 }
 
 func syftSBOMProvider(userInput string, config ProviderConfig) ([]Package, Context, error) {
-	reader, err := getSBOMReader(userInput)
+	s, err := getSBOM(userInput, config)
 	if err != nil {
 		return nil, Context{}, err
 	}
 
-	sbom, format, err := syft.Decode(reader)
+	return FromCatalog(s.Artifacts.PackageCatalog, config), Context{
+		Source: &s.Source,
+		Distro: s.Artifacts.LinuxDistribution,
+	}, nil
+}
+
+func newInputInfo(scheme, contentTye string) *inputInfo {
+	return &inputInfo{
+		Scheme:      scheme,
+		ContentType: contentTye,
+	}
+}
+
+type inputInfo struct {
+	ContentType string
+	Scheme      string
+}
+
+func getSBOM(userInput string, config ProviderConfig) (*sbom.SBOM, error) {
+	reader, info, err := getSBOMReader(userInput, config)
+	if err != nil {
+		return nil, err
+	}
+
+	if info != nil {
+		if (info.Scheme == "sbom" || info.ContentType == "sbom") && config.AttestationKey != "" {
+			return nil, fmt.Errorf("key is meant for atttestation verification, your input is a plain SBOM and doesn't need it")
+		}
+
+		if info.Scheme == "att" && info.ContentType != "att" {
+			return nil, fmt.Errorf("scheme specify an attestation but the content is not an attestation")
+		}
+	}
+
+	s, format, err := syft.Decode(reader)
 	if err != nil {
-		return nil, Context{}, fmt.Errorf("unable to decode sbom: %w", err)
+		return nil, fmt.Errorf("unable to decode sbom: %w", err)
 	}
+
 	if format == nil {
-		return nil, Context{}, errDoesNotProvide
+		return nil, errDoesNotProvide
 	}
 
-	return FromCatalog(sbom.Artifacts.PackageCatalog, config), Context{
-		Source: &sbom.Source,
-		Distro: sbom.Artifacts.LinuxDistribution,
-	}, nil
+	return s, nil
 }
 
-func getSBOMReader(userInput string) (io.Reader, error) {
-	if userInput == "" {
+func getSBOMReader(userInput string, config ProviderConfig) (io.Reader, *inputInfo, error) {
+	switch {
+	// the order of cases matter
+	case userInput == "":
 		// we only want to attempt reading in from stdin if the user has not specified other
 		// options from the CLI, otherwise we should not assume there is any valid input from stdin.
-		return stdinReader(), nil
-	}
+		return decodeStdin(stdinReader(), config)
 
-	if explicitlySpecifyingSBOM(userInput) {
+	case explicitlySpecifyingSBOM(userInput):
 		filepath := strings.TrimPrefix(userInput, "sbom:")
+		r, err := openFile(filepath)
+		return r, newInputInfo("sbom", "sbom"), err
 
-		sbom, err := openSbom(filepath)
+	case explicitlySpecifyAttestation(userInput):
+		path := strings.TrimPrefix(userInput, "att:")
+		f, err := openFile(path)
 		if err != nil {
-			return nil, fmt.Errorf("unable to use specified SBOM: %w", err)
+			return nil, nil, err
 		}
+		r, err := getSBOMFromAttestation(f, config)
+		return r, newInputInfo("att", "att"), err
 
-		if !sbomHasContent(sbom) {
-			return nil, errEmptySBOM{filepath}
+	case isPossibleAttestation(userInput):
```

Comment:
We talked about this on the call but just wanted to capture this for others to see. I think this can return true if an sbom had some description or comment field with the mime type in it.

I think we want to make sure we account for cases where `isPossibleAttestation` and `isPossibleSBOM` both return true. 

If both return true then only one of the resulting workflow will not return an error so we can try both and decide which one to return to the user. If both return an error saying we tried both